packages/serialized-state-manager/src/__tests__/index.test.ts

```diff
@@ -39,36 +36,85 @@ describe('serializedStateManagerDefinition', () => {
   });
 
   describe('SerializedStateManagerV0', () => {
-    let serializedStateManagerBinder: FeatureServiceBinder<
-      SerializedStateManagerV0
-    >;
+    let integratorSerializedStateManager: SerializedStateManagerV0;
+    let consumer1SerializedStateManager: SerializedStateManagerV0;
+    let consumer2SerializedStateManager: SerializedStateManagerV0;
 
     beforeEach(() => {
-      serializedStateManagerBinder = serializedStateManagerDefinition.create(
+      const serializedStateManagerBinder = serializedStateManagerDefinition.create(
         mockEnv
       )['0.1'];
-    });
 
-    it('does not implement anything yet', () => {
-      const serializedStateManager = serializedStateManagerBinder(
-        'test:consumer'
+      integratorSerializedStateManager = serializedStateManagerBinder(
+        'test:integrator'
       ).featureService;
 
-      expect(() =>
-        serializedStateManager.createSerializedStates()
-      ).toThrowError('Method not implemented.');
+      consumer1SerializedStateManager = serializedStateManagerBinder(
+        'test:consumer:1'
+      ).featureService;
 
-      expect(() => serializedStateManager.getSerializedState()).toThrowError(
-        'Method not implemented.'
-      );
+      consumer2SerializedStateManager = serializedStateManagerBinder(
+        'test:consumer:2'
+      ).featureService;
+    });
 
-      expect(() => serializedStateManager.register(jest.fn())).toThrowError(
-        'Method not implemented.'
-      );
+    describe('#serializeStates', () => {
+      describe('when no consumer has called #register', () => {
+        it('returns a stringified empty object', () => {
+          expect(integratorSerializedStateManager.serializeStates()).toBe(
+            encodeURI('{}')
+          );
+        });
+      });
 
-      expect(() => serializedStateManager.setSerializedStates('')).toThrowError(
-        'Method not implemented.'
-      );
+      describe('when consumers have called #register', () => {
+        beforeEach(() => {
+          consumer1SerializedStateManager.register(() => '{kind:"foo"}');
```

Comment:
for better readability maybe use JSON.stringify here?

Reply:
fixed